Sources/NIOKit/Future+Optional.swift

```diff
@@ -0,0 +1,94 @@
+import NIO
+
+extension EventLoopFuture {
+    
+    /// Calls a closure on an optional value in an `EventLoopFuture` if it exists.
+    ///
+    ///     let optional = eventLoop.future(Optiona<Int>.some(42))
+    ///     let some = optional.mapOptional { int -> Float in
+    ///         return int * 3.14
+    ///     }
+    ///     // some: EventLoopFuture(Optional(131.88))
+    ///
+    /// - parameters:
+    ///   - closure: The closure to call on the unwrapped optional value.
+    ///   - unwrapped: The optional's value, unwrapped.
+    /// - returns: The result of the closure if the optional was unwrapped, or nil if it wasn't, wrapped in an `EventLoopFuture`.
+    public func mapOptional<Wrapped, Result>(_ closure: @escaping (_ unwrapped: Wrapped)throws -> Result) -> EventLoopFuture<Result?>
```

Comment:
We should name this `flatMapOptionalThrowing` since it calls `flatMapThrowing` internally. (Even though I disagree with `flatMapThrowing`'s naming in general, we should match NIO)